src/lib/storage/constraints/base_constraint_checker.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <optional>
+#include <string>
+#include <tuple>
+
+#include "storage/chunk.hpp"
+#include "storage/table.hpp"
+
+namespace opossum {
+
+class BaseConstraintChecker {
+ public:
+  BaseConstraintChecker(const Table& table, const TableConstraintDefinition& constraint)
+      : _table(table), _constraint(constraint) {}
+  virtual ~BaseConstraintChecker() = default;
+
+  virtual std::tuple<bool, ChunkID> is_valid(const CommitID snapshot_commit_id, const TransactionID our_tid) = 0;
```

Comment:
I'm of the same opinion as @mrks. I wouldn't leave it out here just because we can, also I didn't know before that this was possible and it would be more confusing than helpful in some cases I think.